### PR TITLE
Rebase master to 4-0-dev and fix build

### PR DESCRIPTION
**Pull request by benoittgt**
Address merge conflict in https://github.com/rspec/rspec-rails/pull/2071

### CODE REVIEW

**Comment by benoittgt** [comment]
Waiting for https://github.com/rspec/rspec-rails/pull/2158

**Comment by benoittgt** [comment]
I will fix the Rubocop offenses tonight if I have time.

**Comment by benoittgt** [comment]
Ok... I think we should maybe unsync rubocop from the rest of RSpec...

At the moment if we jump back to 0.52

```
 rubocop lib                                         [22:45:09]
/Users/bti/code/rspec-dev/repos/rspec-rails/Gemfile:23: warning: already initialized constant Bundler::Dsl::RAILS_VERSION
/Users/bti/code/rspec-dev/repos/rspec-rails/Gemfile:3: warning: previous definition of RAILS_VERSION was here
Warning: unrecognized cop Style/TrailingCommaInArrayLiteral found in .rubocop_rspec_base.yml
Warning: unrecognized cop Style/TrailingCommaInHashLiteral found in .rubocop_rspec_base.yml
Warning: unrecognized cop Layout/IndentFirstArgument found in .rubocop_rspec_base.yml
Warning: unrecognized cop Style/MethodMissingSuper found in .rubocop_rspec_base.yml
Warning: unrecognized cop Style/MissingRespondToMissing found in .rubocop_rspec_base.yml
Warning: unrecognized cop Layout/IndentFirstArrayElement found in .rubocop_rspec_base.yml
Warning: unrecognized cop Style/AccessModifierDeclarations found in .rubocop.yml
Warning: unrecognized cop Naming/MemoizedInstanceVariableName found in .rubocop.yml
Warning: unrecognized cop Naming/UncommunicativeMethodParamName found in .rubocop.yml
Warning: unrecognized cop Naming/RescuedExceptionsVariableName found in .rubocop.yml
Warning: unrecognized cop Style/TrailingCommaInArrayLiteral found in .rubocop.yml
Warning: unrecognized cop Style/TrailingCommaInHashLiteral found in .rubocop.yml
Warning: unrecognized cop Layout/IndentFirstArgument found in .rubocop.yml
Warning: unrecognized cop Style/MethodMissingSuper found in .rubocop.yml
Warning: unrecognized cop Style/MissingRespondToMissing found in .rubocop.yml
Warning: unrecognized cop Layout/IndentFirstArrayElement found in .rubocop.yml
Inspecting 65 files
...................C.............C.C.E...........................

Offenses:
```

Way too many cops have changed. I think the goal of RSpec 4 will be also to jump to the last rubocop like `rspec_rails`.

**Comment by benoittgt** [comment]
CI looks better. For Rails version below 5 it was looking similar to what you fixed Jon. I removed the restriction to `is_ruby_23`.
BTW I saw some interesting doc on travis documentation. https://docs.travis-ci.com/user/languages/ruby/

Capture d’écran 2019-08-26 à 23 50 34

On jruby-head and Rails 6 I commented on an open issue: https://github.com/jruby/activerecord-jdbc-adapter/issues/980 and the super team there help me identifying a wrong version on `activerecord-jdbcsqlite3`. Jumping to last version seems better even if I still have one error that I don't understand yet. I will try to continue tomorrow.

**Comment by benoittgt** [comment]
The failing build is related to last release of Rails `5.2.3.rc1`

```
Bundler could not find compatible versions for gem "actionpack":
  In Gemfile:
    rails (~> 5.2.2) java was resolved to 5.2.3.rc1, which depends on
      actionpack (= 5.2.3.rc1) java

    rspec-rails java was resolved to 4.0.0.beta2, which depends on
      actionpack (>= 4.2) java
```

File: Gemfile-rails-dependencies [hunk 1]
```diff
@@ -33,5 +33,10 @@ else
   if version >= '5-1-stable' && RUBY_VERSION  >= "2.3"
     gem "puma"
   end
-  gem 'activerecord-jdbcsqlite3-adapter', platforms: [:jruby]
+
+  if /(\d+)/.match(ENV['RAILS_VERSION'])[1].to_i >= 6
```
Comment (benoittgt) [inline Gemfile-rails-dependencies]: Not super fan about this. Maybe you have better idea @JonRowe

Comment (JonRowe) [reply Gemfile-rails-dependencies]: We usually do `ENV['RAILS_VERSION'].to_f >= 6.0`  etc but you do sometimes have to stripe the other chars, I'd probably use `.gsub(/[^\d\.]/,'').to_f` or similar

Comment (benoittgt) [reply Gemfile-rails-dependencies]: Thanks! Applied in https://github.com/rspec/rspec-rails/pull/2160/commits/fe90da4ea804ea9a52b0402f601cf9479b83ff99

Happy to merge the PR if you are ok.

**Pull request merged**